### ISSUE TITLE
Proposal: Add a bool isDirty() function

### ISSUE THREAD

**Issue by ScruffR** [ubidots/ubidots-particle#31]
Since `sendAll()` does a load of stuff (mainly the `malloc()` call seems costly) unconditionally it would be good to have a builtin way to retrieve a flag (e.g. `bool isDirty()`) whether there are any pending updates waiting for a `sendAll()` like this
```
  if (Ubidots.isDirty()) 
    Ubidots.sendAll();
  else
    Log.trace("Nothing to send");
```

**Comment by jotathebest** [comment]
Greetings @ScruffR , thanks for your feedback, I think that it would be nice a method like the one that you describe. I will include it in the next update of the library.

Regards

**Comment by jotathebest** [comment]
solved in version 2.1.13
